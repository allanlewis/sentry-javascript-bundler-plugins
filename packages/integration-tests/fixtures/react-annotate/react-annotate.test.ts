import childProcess from "child_process";
import path from "path";

function checkBundle(bundlePath: string): void {
  const processOutput = childProcess.execSync(`node ${bundlePath}`, { encoding: "utf-8" });
  expect(processOutput).toMatchSnapshot();
}

test.todo("esbuild bundle");

test("rollup bundle", () => {
  expect.assertions(1);
  checkBundle(path.join(__dirname, "./out/vite/index.js"));
});

test("vite bundle", () => {
  expect.assertions(1);
  checkBundle(path.join(__dirname, "./out/vite/index.js"));
});

test.todo("webpack 4 bundle if node is < 18");

test.todo("webpack 5 bundle");
