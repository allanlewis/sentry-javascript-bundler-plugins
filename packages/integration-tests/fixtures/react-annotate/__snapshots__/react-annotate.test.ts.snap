// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup bundle 1`] = `
"<div><span data-sentry-component=\\"ComponentA\\" data-sentry-source-file=\\"component-a.jsx\\">Component A</span></div>
"
`;

exports[`vite bundle 1`] = `
"<div><span data-sentry-component=\\"ComponentA\\" data-sentry-source-file=\\"component-a.jsx\\">Component A</span></div>
"
`;

exports[`webpack 5 bundle 1`] = `
"<div><span data-sentry-component=\\"ComponentA\\" data-sentry-source-file=\\"component-a.jsx\\">Component A</span></div>
"
`;
